src/pybind/mgr/dashboard/frontend/e2e/page-helper.po.ts

```diff
@@ -21,6 +21,80 @@ export abstract class PageHelper {
     return $$('.nav.nav-tabs li').count();
   }
 
+  /**
+   * Returns the cell with the exact content given in `content`.
+   *
+   * Limitations: Does not work correctly if the table contains more than 10
+   * entries.
+   */
+  static getTableCellByContent(content: string): ElementFinder {
+    const elem = $('datatable-footer');
```

Comment:
If we're worried about the pool being too deep into the entry list and not appearing on the first page how do you like the idea of using the search bar to filter the entry list? Essentially instead of checking the number of pools and such we have a line like:

```Typescript

$('input.form-control.ng-valid').sendKeys(content);

```
That would filter the results to only show entries containing the name of the pool, even entries that weren't originally on page 1. The only downside is that this requires us to give our pools somewhat more unique names. For example, this would not work with a pool called 'a' but if you named the pool 'unique' or 'testpool' I think this solution would work regardless of the number of entries in the table.

Reply:
I tend to increase the entries shown in the table to 100. That's also not perfect, as it's limited to 100 entries and would still find other table cells with the same content, but I think it'll be more precise. Unless the search field can be used to precisely match a table cell. In that case, that'd would be the better approach. I'll look into it.

Reply:
We should make sure the object we create for testing has as unique name, every test file should have a different prefix. 
p.e.: `pool_e2e_pool1` vs `rbd_e2e_pool1`
Then we can filter the table for `pool_e2e` or `rbd_e2e` and get the correct values.

We should create a tracker issue for this.
In the corresponding PR we should fix this in all existing tests and improve HACKING with this information.

Reply:
Laura has proposed the same, but I was unhappy with the inaccuracy of the search, if it's not possible to search for exact content (still don't know). Nonetheless, you seem to have formed a majority, so I will just implement it as proposed and if there are any limitations with the implementation, I will just raise appropriate developer-friendly exceptions.